ckan/config/middleware/__init__.py

```diff
@@ -34,3 +39,37 @@ def make_app(conf: Union[Config, CKANConfig]) -> CKANApp:
     _internal_test_request_context = flask_app._wsgi_app.test_request_context()
 
     return flask_app
+
+
+class CKANSecureCookieSessionInterface(SecureCookieSessionInterface):
+    """Flask cookie-based sessions with expiration support.
+    """
+
+    def __init__(self, app: CKANApp):
+        pass
+
+    def open_session(self, app: CKANApp, request: Request):
+        session = super().open_session(app, request)
+        if session:
+            # Cookie-based sessions expire with the browser's session. The line
+            # below changes this behavior, extending session's lifetime by
+            # `PERMANENT_SESSION_LIFETIME` seconds. `SESSION_PERMANENT` option
+            # is used as indicator of permanent sessions by flask-session
+            # package, so we also should rely on it, for predictability.
+            session.setdefault("_permanent", app.config["SESSION_PERMANENT"])
+
+        return session
+
+
+class CKANRedisSessionInterface(RedisSessionInterface):
+    """Flask-Session redis-based sessions with CKAN's Redis connection.
+    """
```

Comment:
I would document here as well why do we need this level of customization. (For future maintainers)

Reply:
Thanks, good catch